### PR TITLE
Bullhorn REST SDK v2.0

### PR DESCRIPTION
**Pull request by scriptom**
Major changes include:
- Support for Java 17
- Adding new fields to entities
- Adding new entities

### CODE REVIEW

File: src/main/java/com/bullhornsdk/data/model/entity/association/standard/CandidateAssociations.java [hunk 1]
```diff
@@ -46,6 +37,7 @@ public final class CandidateAssociations implements EntityAssociations<Candidate
     private final AssociationField<Candidate, PersonCustomObjectInstance8> customObject8s = instantiateAssociationField("customObject8s", PersonCustomObjectInstance8.class);
     private final AssociationField<Candidate, PersonCustomObjectInstance9> customObject9s = instantiateAssociationField("customObject9s", PersonCustomObjectInstance9.class);
     private final AssociationField<Candidate, PersonCustomObjectInstance10> customObject10s = instantiateAssociationField("customObject10s", PersonCustomObjectInstance10.class);
+    private final AssociationField<Candidate, PersonCustomObjectInstance29> customObject29s = instantiateAssociationField("customObject29s", PersonCustomObjectInstance29.class);
```
Comment (angusyyl) [inline src/main/java/com/bullhornsdk/data/model/entity/association/standard/CandidateAssociations.java:40]: Is there a reason why we don't include all the other customObject11s to customObject28s? And customObject30 should be already added in #410 but now is gone.

File: src/main/java/com/bullhornsdk/data/model/entity/core/customobjectinstances/person/PersonCustomObjectInstance29.java
Comment (angusyyl) [inline src/main/java/com/bullhornsdk/data/model/entity/core/customobjectinstances/person/PersonCustomObjectInstance29.java:1]: Can we add PersonCustomObjectInstance26 to PersonCustomObjectInstance28 as well?

File: src/main/java/com/bullhornsdk/data/model/entity/core/standard/Appointment.java [hunk 2]
```diff
@@ -104,6 +101,8 @@ public class Appointment extends AbstractEntity implements QueryEntity, UpdateEn
 
     private String timeZoneID;
 
+    private JobSubmission jobSubmission;
```
Comment (angusyyl) [inline src/main/java/com/bullhornsdk/data/model/entity/core/standard/Appointment.java:104]: Is this missing in the getter, setter, hashCode, etc.?

File: src/main/java/com/bullhornsdk/data/model/entity/core/standard/Candidate.java [hunk 3]
```diff
@@ -53,7 +56,10 @@
 		"submissions", "ssn", "stateAddtionalWitholdingsAmount", "stateExemptions", "stateFilingStatus", "status", "tasks", "taxID",
 		"taxState", "tearsheets", "timeZoneOffsetEST", "travelLimit", "type", "username", "veteran", "webResponses", "willRelocate", "workAuthorized",
 		"workHistories", "workPhone", "customObject1s", "customObject2s", "customObject3s", "customObject4s", "customObject5s", "customObject6s",
-        "customObject7s", "customObject8s", "customObject9s", "customObject10s" })
+    "customObject7s", "customObject8s", "customObject9s", "customObject10s", "customObject29s", "activePlacements",
```
Comment (angusyyl) [inline src/main/java/com/bullhornsdk/data/model/entity/core/standard/Candidate.java:59]: Can we add customObject11s to customObject28s and customObject30s as well?

File: src/main/java/com/bullhornsdk/data/model/entity/core/standard/CorporateUser.java [hunk 4]
```diff
@@ -159,6 +160,22 @@ public class CorporateUser extends CustomFieldsA implements QueryEntity, Associa
...
+    private DateTime dateLastModified;
+
+    private OneToMany<CorporateUser> delegations;
+
+    private String departmentIdList;
+
+    private OneToMany<DistributionList> distributionLists;
+
+    private OneToMany<DistributionList> favoriteSavedSearches;
+
```
Comment (angusyyl) [inline src/main/java/com/bullhornsdk/data/model/entity/core/standard/CorporateUser.java:178]: Can we also include corporation field?

File: src/main/java/com/bullhornsdk/data/model/entity/core/standard/CorporateUser.java [hunk 5]
```diff
@@ -159,6 +160,22 @@ public class CorporateUser extends CustomFieldsA implements QueryEntity, Associa
 
     private DateTime userDateAdded;
 
+    private Location addressSourceLocation;
```
Comment (angusyyl) [inline src/main/java/com/bullhornsdk/data/model/entity/core/standard/CorporateUser.java:163]: Do we need to add these new fields in @JsonPropertyOrder, getters, setters, etc?

File: src/main/java/com/bullhornsdk/data/model/entity/core/standard/Opportunity.java [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
-import com.bullhornsdk.data.model.entity.core.customobjectinstances.joborder.JobOrderCustomObjectInstance2;
-import com.bullhornsdk.data.model.entity.core.customobjectinstances.joborder.JobOrderCustomObjectInstance3;
-import com.bullhornsdk.data.model.entity.core.customobjectinstances.joborder.JobOrderCustomObjectInstance4;
-import com.bullhornsdk.data.model.entity.core.customobjectinstances.joborder.JobOrderCustomObjectInstance5;
-import com.bullhornsdk.data.model.entity.core.customobjectinstances.joborder.JobOrderCustomObjectInstance6;
-import com.bullhornsdk.data.model.entity.core.customobjectinstances.joborder.JobOrderCustomObjectInstance7;
-import com.bullhornsdk.data.model.entity.core.customobjectinstances.joborder.JobOrderCustomObjectInstance8;
-import com.bullhornsdk.data.model.entity.core.customobjectinstances.joborder.JobOrderCustomObjectInstance9;
+import com.bullhornsdk.data.api.helper.json.DynamicNullValueFilter;
+import com.bullhornsdk.data.model.entity.core.customobjectinstances.opportunity.OpportunityCustomObjectInstance1;
```
Comment (angusyyl) [inline src/main/java/com/bullhornsdk/data/model/entity/core/standard/Opportunity.java:5]: Good catch!

File: src/main/java/com/bullhornsdk/data/model/enums/BullhornEntityInfo.java [hunk 7]
```diff
@@ -456,6 +460,7 @@ public enum BullhornEntityInfo {
     PERSON_CUSTOM_OBJECT_INSTANCE_23("PersonCustomObjectInstance23", PersonCustomObjectInstance23.class, PersonCustomObjectInstance23Wrapper.class, PersonCustomObjectInstance23ListWrapper.class, "PersonCustomObject23EditHistory", "PersonCustomObjectEditHistoryFieldChange"),
     PERSON_CUSTOM_OBJECT_INSTANCE_24("PersonCustomObjectInstance24", PersonCustomObjectInstance24.class, PersonCustomObjectInstance24Wrapper.class, PersonCustomObjectInstance24ListWrapper.class, "PersonCustomObject24EditHistory", "PersonCustomObjectEditHistoryFieldChange"),
     PERSON_CUSTOM_OBJECT_INSTANCE_25("PersonCustomObjectInstance25", PersonCustomObjectInstance25.class, PersonCustomObjectInstance25Wrapper.class, PersonCustomObjectInstance25ListWrapper.class, "PersonCustomObject25EditHistory", "PersonCustomObjectEditHistoryFieldChange"),
+    PERSON_CUSTOM_OBJECT_INSTANCE_29("PersonCustomObjectInstance29", PersonCustomObjectInstance29.class, PersonCustomObjectInstance29Wrapper.class, PersonCustomObjectInstance29ListWrapper.class, "PersonCustomObject29EditHistory", "PersonCustomObjectEditHistoryFieldChange"),
```
Comment (angusyyl) [inline src/main/java/com/bullhornsdk/data/model/enums/BullhornEntityInfo.java:463]: Can we include PERSON_CUSTOM_OBJECT_INSTANCE_26 to PERSON_CUSTOM_OBJECT_INSTANCE_28 and PERSON_CUSTOM_OBJECT_INSTANCE_30?

File: src/main/java/com/bullhornsdk/data/model/entity/core/standard/Placement.java [hunk 8]
```diff
@@ -1404,6 +1722,7 @@ public String toString() {
             ", commissions=" + commissions +
```
Comment (taswartz) [inline src/main/java/com/bullhornsdk/data/model/entity/core/standard/Placement.java:1722]: @scriptom 
We missed adding all of the new fields to the 'toString()' and 'hashCode()' functions

**Review by johnsully83 — APPROVED** [review]

**Pull request merged**